Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.h

```diff
@@ -22,11 +22,11 @@ namespace AZ
 
 //! Macros to self-register AutoGen node into ScriptCanvas
 #define REGISTER_SCRIPTCANVAS_AUTOGEN_FUNCTION(LIBRARY)\
-    static ScriptCanvas::LIBRARY##FunctionRegistry s_AutoGenFunctionRegistry;
+    static ScriptCanvas::LIBRARY##FunctionRegistry s_##LIBRARY##AutoGenFunctionRegistry;
 #define REGISTER_SCRIPTCANVAS_AUTOGEN_NODEABLE(LIBRARY)\
-    static ScriptCanvas::LIBRARY##NodeableRegistry s_AutoGenNodeableRegistry;
+    static ScriptCanvas::LIBRARY##NodeableRegistry s_##LIBRARY##AutoGenNodeableRegistry;
 #define REGISTER_SCRIPTCANVAS_AUTOGEN_GRAMMAR(LIBRARY)\
-    static ScriptCanvas::LIBRARY##GrammarRegistry s_AutoGenGrammarRegistry;
+    static ScriptCanvas::LIBRARY##GrammarRegistry s_##LIBRARY##AutoGenGrammarRegistry;
```

Comment:
each of these should be an singleton object, why do you need to give then different name?

Reply:
It's feasible that someone could have multiple libraries in one gem, currently this would result in 'redefinition of s_AutoGenNodeableRegistry' compilation errors. With this change these are still singletons, but their unique names prevent compilation errors.